tests/test_render_validity.py

```
@@ -297,3 +297,80 @@ def test_docs_on_by_default_keeps_sphinx_subsystem(
+    assert job["strategy"]["matrix"]["include"] == [
+        {"os": "ubuntu-latest", "tox_args": ""},
+        {"os": "macos-latest", "tox_args": "-e py"},
+        {"os": "windows-latest", "tox_args": "-e py"},
+    ]
+    assert (
+        _ci_test_step(job)["run"] == "uv run --locked tox run ${{ matrix.tox_args }}"
+    )
+
+
+def test_c_extensions_restores_the_full_grid(render: Callable[..., Path]) -> None:
+    """The one carve-out: a compiled extension makes every OS × interpreter pair a
```

Ruff (0.16.1)

[warning] Docstring contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF002)